src/PAModel/IR/IRStateHelpers.cs

```diff
@@ -503,15 +503,18 @@ private static ControlInfoJson.DynamicPropertyJson GetDynamicPropertyEntry(Contr
             DynamicPropertyState propState = null;
             if (state.DynamicProperties.ToDictionary(prop => prop.PropertyName).TryGetValue(propName, out propState))
             {
-                property.Rule = new ControlInfoJson.RuleEntry()
+                if (propState.Property != null)
```

Comment:
Add a comment about how the DynamicProperties list may contain items without a corresponding property, and those correspond to empty properties, rather than properties with a default value.